src/liboslcomp/oslcomp.cpp

```diff
@@ -1149,8 +1158,30 @@ OSLCompilerImpl::check_write_legality (const Opcode &op, int opnum,
                sym->name().c_str(), opnum);
     }
 
-    // FIXME -- check for writing to globals.  But it's tricky, depends on
-    // what kind of shader we are.
+    // Check for writing to globals.
+    if (sym->symtype() == SymTypeGlobal) {
+        static ustring Ci("Ci"), P("P"), I("I"), N("N"), Ng("Ng");
+        static ustring dPdu("Pdu"), dPdv("Pdv"), u("u"), v("v");
+        static ustring time("time"), dtime("dtime"), dPdtime("dPdtime");
+        ShaderType stype = shadertype();
+        ustring symname = sym->name();
+        bool ok = true;
```

Comment:
Generic shaders should be `ok=false` ?

Reply:
At least in testsuite, we seem to depend on generic shading being able to do anything.

Reply:
Is that a bug or a feature? Might be beyond the scope of this review, but I think it would be worth clamping down on this.

Reply:
I don't know if it's a bug or feature. I'm going to open a separate discussion about this on osl-dev, there's an important question of future direction here and I want a diversity of opinions.